controllers/openstackdataplane_controller.go

```diff
@@ -133,6 +135,7 @@ func (r *OpenStackDataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 	var deployErrors []string
 	shouldRequeue := false
 	if instance.Spec.DeployStrategy.Deploy {
+
```

Comment:
Add the logic here as well to set the DeployIdentifier if not set.

Reply:
At line 170, instead of just setting DeployStrategy.Deploy on the role, set DeployStrategy. That way the role gets the same DeployIdentifier as the DataPlane, and the role won't generate a new one.

Reply:
Before line 166, we need to add a check if the role IsReady, and if the DeployIdentifier on the role matches that of the DataPlane. If both are true, then that means the role completed the deploy associated with this DataPlane deploy request. If that's the case, don't set DeployStrategy on the role again.

I think that's what's causing the loop....the role finishes and sets it's deploy=false, then from here it's get set back to true.

Reply:
Hey @slagle, i tried this approach but it didn't work. For what I saw the role deployIdentifer and the dataplane deployIdentifier are always different...i guess because we never pass "deployStrategy" from dataplane to dataplanerole (in the CreateOrPatch).

Reply:
It looks like it's doing the right thing to me. I used some of the kuttl test files to test it:


```shell
Mon Jun 05 02:36:23 (jslagle@teletran-1) [~/code/osp-ng/openstack-k8s-operators/dataplane-operator] (deployIdentifier)$ oc create -f tests/kuttl/tests/dataplane-deploy-no-nodes-test/00-dataplane-create.yaml 
openstackdataplane.dataplane.openstack.org/openstack-edpm-no-nodes created

Mon Jun 05 03:51:31 (jslagle@teletran-1) [~/code/osp-ng/openstack-k8s-operators/dataplane-operator] (deployIdentifier)$ oc apply -f tests/kuttl/tests/dataplane-deploy-no-nodes-test/01-dataplane-deploy.yaml 
openstackdataplane.dataplane.openstack.org/openstack-edpm-no-nodes configured


Mon Jun 05 03:51:59 (jslagle@teletran-1) [~/code/osp-ng/osp-ng] (main)$ oc get osdp openstack-edpm-no-nodes -o json | jq .spec.deployStrategy
{
  "deploy": true,
  "deployIdentifier": "31822b62-c984-4ee7-9d17-73feb48a66ea"
}

Mon Jun 05 03:52:40 (jslagle@teletran-1) [~/code/osp-ng/osp-ng] (main)$ oc get osdprole edpm-compute-no-nodes -o json | jq .spec.deployStrategy
{
  "deploy": true,
  "deployIdentifier": "31822b62-c984-4ee7-9d17-73feb48a66ea"
}

```

The other thing I am seeing though is an infinite reconcile loop between DataPlane and the role. Looks like the DataPlane keeps trying to toggle deploy back to true on the role.